dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt

```diff
@@ -53,58 +63,152 @@ class DataManager(
     }
 
     /**
-     * Method to make the data manager add data to a data store and store meta data in Dataland
+     * Method to make the data manager add data to a data store, store metadata in Dataland and sending messages to the
+     * relevant message queues
      * @param storableDataSet contains all the inputs needed by Dataland
      * @return ID of the newly stored data in the data store
      */
     @Transactional
-    fun addDataSet(storableDataSet: StorableDataSet, correlationId: String): String {
+    fun addDataSetToTemporaryStorageAndSendMessage(storableDataSet: StorableDataSet, correlationId: String):
+        String {
         val company = companyManager.getCompanyById(storableDataSet.companyId)
         logger.info(
             "Sending StorableDataSet of type ${storableDataSet.dataType} for company ID " +
                 "${storableDataSet.companyId}, Company Name ${company.companyName} to storage Interface. " +
                 "Correlation ID: $correlationId",
         )
-        val dataId: String = storeDataSet(storableDataSet, company.companyName, correlationId)
-        metaDataManager.storeDataMetaInformation(
-            dataId,
-            storableDataSet.dataType,
-            storableDataSet.uploaderUserId,
-            storableDataSet.uploadTime,
-            company,
+        val dataId: String = storeDataSetInTemporaryStoreAndSendMessage(
+            storableDataSet, company.companyName, correlationId,
+        )
+        val metaData = DataMetaInformationEntity(
+            dataId, storableDataSet.dataType.toString(),
+            storableDataSet.uploaderUserId, storableDataSet.uploadTime, company, QAStatus.Pending,
         )
+        metaDataManager.storeDataMetaInformation(metaData)
         return dataId
     }
 
-    private fun storeDataSet(
+    /**
+     * Method that listens to the qa_queue and updates the metadata information after successful qa process
+     * @param message is the message delivered on the message queue
+     */
+    @RabbitListener(
+        bindings = [
+            QueueBinding(
+                value = Queue("dataStoredBackendDataManager"),
+                exchange = Exchange(MqConstants.dataStored, declare = "false"),
+                key = [""],
+            ),
+        ],
+    )
+    fun updateMetaData(message: Message) {
```

Comment:
This method binds to the wron exchange, doesn't it? It should bit to MqConstants.dataQualityAssured

Reply:
Yes it was wrong. I corrected it.

Reply:
ok